autorest/codegen/serializers/general_serializer.py

```diff
@@ -20,12 +20,27 @@ def serialize_pkgutil_init_file(self) -> str:
 
     def serialize_init_file(self) -> str:
         template = self.env.get_template("init.py.jinja2")
-        return template.render(code_model=self.code_model, async_mode=self.async_mode,)
+        return template.render(code_model=self.code_model, async_mode=self.async_mode)
+
+    def _correct_credential_parameter(self):
+        credential_param = [
+            gp for gp in self.code_model.global_parameters.parameters if isinstance(gp.schema, CredentialSchema)
+        ][0]
+        credential_param.schema = CredentialSchema(async_mode=self.async_mode)
+        # we know that if there's a credential global param, it's always the first because
+        # we insert it there
+        self.code_model.global_parameters[0] = credential_param
 
     def serialize_service_client_file(self) -> str:
         template = self.env.get_template("service_client.py.jinja2")
+
+        global_parameters = self.code_model.global_parameters
+        if self.code_model.options['credential']:
+            self._correct_credential_parameter()
+
         return template.render(
             code_model=self.code_model,
+            global_parameters=global_parameters,
```

Comment:
Why the need for a new parameter passed? We can still access `code_model.global_parameters`

Reply:
this was a remnant of my old way of doing that, will change back, thanks for catching!

Reply:
@iscai-msft still needs fix ;)